vdr/doc-creator.go

```diff
@@ -0,0 +1,112 @@
+package vdr
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/shengdoushi/base58"
+
+	"github.com/nuts-foundation/go-did"
+
+	nutsCrypto "github.com/nuts-foundation/nuts-node/crypto"
+)
+
+// NutsDIDMethodName is the DID method name used by Nuts
+const NutsDIDMethodName = "nuts"
+
+// NutsDocCreator implements the DocCreator interface and can create Nuts DID Documents.
+type NutsDocCreator struct {
+	// keyCreator is used for getting a fresh key and use it to generate the Nuts DID
+	keyCreator nutsCrypto.KeyCreator
+}
+
+func didKidNamingFunc(pKey crypto.PublicKey) (string, error) {
```

Comment:
some missing coverage

Reply:
As far as I can see the only coverage that is missing is a failing jwk.New and jwk.AssignKeyID. These are not likely to happen since the cast to the correct type that has to succeed. I added a test that checks for an empty curve. That should make it a bit more fool proof.